packages/react/src/components/Table/TableFoot/TableFoot.jsx

```diff
@@ -50,46 +51,57 @@ const defaultProps = {
 
 const TableFoot = ({
   testId,
-  options: { hasRowExpansion, hasRowSelection, hasRowActions },
+  options: { hasRowExpansion, hasRowSelection, hasRowActions, hasRowNesting },
   tableState: { aggregations, ordering },
   showExpanderColumn,
 }) => {
   const visibleColumns = ordering.filter((col) => !col.isHidden);
 
+  const hasMultiSelect = hasRowSelection === 'multi';
+  const hasExpandOrNest = hasRowExpansion || hasRowNesting;
+  const labelColSpan =
+    hasMultiSelect && hasExpandOrNest ? 3 : hasExpandOrNest || hasMultiSelect ? 2 : 1;
+
   return (
     <tfoot className={`${iotPrefix}-table-foot`} data-testid={testId}>
-      <TableRow>
-        {hasRowExpansion ? <TableCell key="row-expansion" /> : null}
-        {hasRowSelection === 'multi' ? <TableCell key="row-selection" /> : null}
+      <TableRow key="aggregate-row">
         {visibleColumns.map((orderedCol, index) => {
           const aggregated = aggregations.columns.find((col) => orderedCol.columnId === col.id);
           const isLabelCell = !aggregated && index === 0;
           const cellTestId = `${testId}-aggregation-${orderedCol.columnId}`;
           const cellKey = `${orderedCol.columnId}${index}`;
 
           return isLabelCell ? (
-            <TableCell
-              className={`${iotPrefix}-table-foot--label`}
-              data-testid={cellTestId}
-              key={cellKey}
-            >
-              {aggregations.label}
-            </TableCell>
+            <Fragment key="label-cell-fragment">
+              <TableCell
+                className={`${iotPrefix}-table-foot--label`}
+                data-testid={cellTestId}
+                key={cellKey}
+                colSpan={labelColSpan}
+              >
+                {aggregations.label}
+              </TableCell>
+            </Fragment>
           ) : aggregated ? (
-            <TableCell
-              className={classnames({
-                [`${iotPrefix}-table-foot--value`]: true,
-                'data-table-end': aggregated.align === 'end',
-                'data-table-start': !aggregated.align || aggregated.align === 'start',
-                'data-table-center': aggregated.align === 'center',
-                [`${iotPrefix}-table-foot--value__sortable`]: aggregated.isSortable,
-              })}
-              align={aggregated.align ? aggregated.align : undefined}
-              data-testid={cellTestId}
-              key={cellKey}
-            >
-              {aggregated.value}
-            </TableCell>
+            <Fragment key={`aggregated-cell-fragment-${index}`}>
+              {index === 0 && (hasMultiSelect || hasExpandOrNest) ? (
+                <TableCell colSpan={hasMultiSelect && hasExpandOrNest ? 2 : 1} />
```

Comment:
nit: this could also be simplified into just to having a single condition to add the extra table cell or not.

Reply:
given that all of this is going to be reworked to account for aggregations in the first column or potentially multiple labels, I don't really think that's important right now.